### ISSUE TITLE
Query: Accesing intermediate layer features

### ISSUE THREAD

**Issue by tridivb** [facebookresearch/SlowFast#24]
Hello Team,

Thank you for releasing the great framework. 

I am trying to train the model on a different dataset and was thinking of extracting the i3d or slowfast features (output after the average pooling layer) once that is done. As I understand, this could be done by tweaking the code in ./slowfast/build/lib/slowfast/models/head_helper.py  

```
 def forward(self, inputs):
        assert (
            len(inputs) == self.num_pathways
        ), "Input tensor does not contain {} pathway".format(self.num_pathways)
        pool_out = []
        for pathway in range(self.num_pathways):
            m = getattr(self, "pathway{}_avgpool".format(pathway))
            pool_out.append(m(inputs[pathway]))
        x = torch.cat(pool_out, 1)
        # (N, C, T, H, W) -> (N, T, H, W, C).
        x = x.permute((0, 2, 3, 4, 1))
        feat = x # save the feature
        # Perform dropout.
        if hasattr(self, "dropout"):
            x = self.dropout(x)
        x = self.projection(x)

        # Performs fully convlutional inference.
        if not self.training:
            x = self.act(x)
            x = x.mean([1, 2, 3])

        x = x.view(x.shape[0], -1)
        return x, feat
```

Something like the above. But is there already an existing way to do this?

**Comment by takatosp1** [comment]
Hi thanks for using the framework!

If you are thinking of training a linear classifier on top of our pretrained feature, you can inherit our header, and  have an additional detach() before the forward. We design the codebase in the way that we are not only wish you extract features with the codebase, but also encourage you train your own task with the codebase! 
I can share a feature extractor helper if you like to use it, but I feel we you have in the code is sufficient!
You are also very welcomed to commit your dataset loader to the codebase!

**Comment by tridivb** [comment]
Thank you for the reply. Inheriting should work fine.

**Comment by tridivb** [comment]
@takatosp1 

Would it be helpful to add a feature extraction option to the model config? In my case I use the features for further tasks like anticipation.

Then I will create a commit and open a pull  request.

**Comment by takatosp1** [comment]
Hi,

Thanks for using the code! Welcome to contribute your code to the codebase!

**Comment by takatosp1** [comment]
Hi @tridivb 

Welcome to contribute your code to the codebase!
